src/services/user/registration.js

```diff
@@ -40,166 +42,246 @@ const populateUser = (app, data) => {
-		if ((user.roles || []).includes('student')) {
-			// wrong birthday object?
-			if (user.birthday instanceof Date && Number.isNaN(user.birthday)) {
-				return Promise.reject(new errors.BadRequest(
-					'Fehler bei der Erkennung des ausgewählten Geburtstages.'
-						+ ' Bitte lade die Seite neu und starte erneut.',
-				));
-			}
-			// wrong age?
-			const age = getAge(user.birthday);
-			if (data.parent_email && age >= CONSENT_WITHOUT_PARENTS_MIN_AGE_YEARS) {
-				return Promise.reject(new errors.BadRequest(
-					`Schüleralter: ${age} Im Elternregistrierungs-Prozess darf der Schüler`
-						+ `nicht ${CONSENT_WITHOUT_PARENTS_MIN_AGE_YEARS} Jahre oder älter sein.`,
-				));
-			} if (!data.parent_email && age < CONSENT_WITHOUT_PARENTS_MIN_AGE_YEARS
-				&& !consentSkipCondition.includes('student')) {
-				return Promise.reject(new errors.BadRequest(
-					`Schüleralter: ${age} Im Schülerregistrierungs-Prozess darf der Schüler`
-						+ ` nicht jünger als ${CONSENT_WITHOUT_PARENTS_MIN_AGE_YEARS} Jahre sein.`,
-				));
 			}
-		}
 
-		// identical emails?
-		if (data.parent_email && data.parent_email === data.email) {
-			return Promise.reject(new errors.BadRequest(
-				'Bitte gib eine unterschiedliche E-Mail-Adresse für dein Kind an.',
-			));
-		}
+			// identical emails?
+			if (data.parent_email && data.parent_email === data.email) {
+				return Promise.reject(
+					new errors.BadRequest(
+						'Bitte gib eine unterschiedliche E-Mail-Adresse für dein Kind an.',
+					),
+				);
+			}
 
-		if (data.password_1 && data.passwort_1 !== data.passwort_2) {
-			return new errors.BadRequest('Die Passwörter stimmen nicht überein');
-		}
-		return Promise.resolve();
-	})
+			if (data.password_1 && data.passwort_1 !== data.passwort_2) {
+				return new errors.BadRequest(
+					'Die Passwörter stimmen nicht überein',
+				);
+			}
+			return Promise.resolve();
+		})
 		.then(() => {
 			const email = data.parent_email || data.student_email || data.email;
 			const { pin } = data;
-			return app.service('registrationPins').find({
-				query: { pin, email },
-			}).then((result) => {
-				// check pin
-				if (result.data.length !== 1 || result.data[0].verified !== true) {
-					return Promise.reject(new Error('Der eingegebene Code konnte leider nicht verfiziert werden. Versuch es doch noch einmal.'));
-				}
-				return Promise.resolve();
-			}).catch((err) => {
-				const msg = err.message || 'Fehler wärend der Pin Überprüfung.';
-				logger.error(msg, err);
-				return Promise.reject(new Error(msg));
-			});
+			return app
+				.service('registrationPins')
+				.find({
+					query: { pin, email },
+				})
+				.then((result) => {
+					// check pin
+					if (
+						result.data.length !== 1 ||
+						result.data[0].verified !== true
+					) {
+						return Promise.reject(
+							new Error(
+								'Der eingegebene Code konnte leider nicht verfiziert werden. Versuch es doch noch einmal.',
```

Comment:
Codacy found an issue: [This line has a length of 123. Maximum allowed is 120.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5987920)